### ISSUE TITLE
Order Details: Nav to next order via "Next" button

### ISSUE THREAD

**Issue by bummytime** [woocommerce/woocommerce-ios#259]
In the upper right corner of the Order Details nav bar, there is a "Next" button:

<img width="733" alt="44369738-142b4180-a49d-11e8-89b1-e7649b384c84" src="https://user-images.githubusercontent.com/154014/44425107-90cb2800-a550-11e8-97f3-e53d9a78bffe.png">

which allows the user to progress to the next order's detail screen (thus not navigating back to the Order List screen).

Ideally, if there is no "next order", this button should either be disabled or hidden complete. Perhaps @folletto could offer insight here. 😄

**Comment by bummytime** [comment]
@kyleaparker Quick question for you regarding this feature. Instead of having just a next button, what about having something like this? (Shamelessly stealing from OmniFocus 3 for iOS here)

<img width="568" alt="napkin 120 02-18-19 1 44 17 pm" src="https://user-images.githubusercontent.com/154014/52973393-967cbb80-3383-11e9-8d4a-19a28dd41ea7.png">

This would allow the user to go either way (up/down) while staying on the order detail screen. Also if you are at the top/bottom of the list the appropriate arrow is disabled like so:

![img_052bf665a140-1](https://user-images.githubusercontent.com/154014/52973511-f4a99e80-3383-11e9-979d-6fb219de7843.jpeg)

There would be a few things to figure out here given the way we load orders with our infinite scrolling/pagination logic, but I think it might be a useful feature potentially. 

🤷‍♂️

**Comment by kyleaparker** [comment]
>@kyleaparker Quick question for you regarding this feature. Instead of having just a next button, what about having something like this? (Shamelessly stealing from OmniFocus 3 for iOS here)

I like this idea! I have a few thoughts: 
- I wonder if this would feel cluttered with the back arrow (back to orders) and up/down arrows
- How much do we currently know about the use case for navigating between orders?
- A less discoverable option could be to swipe between orders

**Comment by bummytime** [comment]
> I wonder if this would feel cluttered with the back arrow (back to orders) and up/down arrows

It probably would. :-)

> How much do we currently know about the use case for navigating between orders?

@folletto would be the person to ask because this was part of his original design. 😄 

> A less discoverable option could be to swipe between orders

Agree. It also _might_ be be confusing because swiping to the right (in non-RTL locales) typically takes you to the previous screen in the navigation stack. So in our case, swiping right on an order detail takes you back to the order list.

**Comment by folletto** [comment]
I recall in the research we identified that people went through the list in sequence: do one order, move to the next, do the next, move to the next, etc.

I choose specifically the design above:

1. It's explicit: "Next" is a word, and while not super explicit, it's better than an invisible control or just an icon
2. iOS HIG doesn't support TWO items in the corner, yet the main use case is to move next so not a biggie
3. I avoided any swiping option because of discoverability, but also because on iOS swipe from the edge is already used to go back, and swipe in the middle seems risky as we might want to use that for something more important later and a minor feature like this doesn't seem worth the risk

I hope it helps :)
